plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/LoggingInvoker.java

```diff
@@ -137,6 +138,23 @@ private void maybeRecord(Class<?> clazz, Supplier<String> message) {
         Class<?> clazz = classOf(lhs);
         maybeRecord(clazz, () -> clazz.getName() + "." + name);
         delegate.setProperty(lhs, name, value);
+        if (value != null) {
+            var owner = findOwner(lhs);
+            if (owner instanceof CpsScript) {
```

Comment:
Seems to work as desired also in libraries: assigning to an implicit field of a `vars/*.groovy` produces a warning, while assigning to a field of a named class in `src/**/*.groovy` does not.